src/Education.UnitTests/XamlGenerator/RawTextProcessorTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Education.XamlGenerator;
+
+namespace SonarLint.VisualStudio.Education.UnitTests
+{
+    [TestClass]
+    public class RawTextProcessorTests
+    {
+        [TestMethod]
+        public void ProcessRawText_OneCrossReference_WithSurroundingText_ReturnsListWithCorrectItems()
+        {
+            string text = " {rule:javascript:S1656} - Implements a check on";
+
+            var list = RawTextProcessor.ProcessRawText(text);
+
+            list.Count.Should().Be(3);
```

Comment:
Minor: there's a FluentAssertion for collections `Should().HaveCount(x)`.

I think there is a diagnostic analyzer that complains if we don't use it.